packages/PowerGrid/src/package.ts

```diff
@@ -253,19 +260,50 @@ export function demoCellTypes() {
 //tags: scWebGPURender
 //input: dynamic sc
 //input: bool show
-export function _scWebGPURender(sc: DG.ScatterPlotViewer, show: boolean) {
-  return scWebGPURender(sc, show);
+export async function _scWebGPURender(sc: DG.ScatterPlotViewer, show: boolean) {
+  try {
+    await scWebGPURender(sc, show);
+  } catch (error) {
+    gpuErrorCounter++;
+  }  
 }
 
 //tags: scWebGPUPointHitTest
 //input: dynamic sc
 //input: dynamic pt
 //output: int result
 export async function _scWebGPUPointHitTest(sc: DG.ScatterPlotViewer, pt: DG.Point) {
-  let result = await scWebGPUPointHitTest(sc, pt);
+  let result = -1;
+  try {
+    result = await scWebGPUPointHitTest(sc, pt);  
+  } catch (error) {
+    gpuErrorCounter++;
+    throw error;
+  }
+  
   return result;
 }
 
+//tags: isWebGPUAvailable
+//output: bool result
+export function isWebGPUAvailable(sc: DG.ScatterPlotViewer) {
+  return gpuDevice != null && !gpuErrorCounter;
+}
+
+//tags: isWebGPURenderValid
+//input: dynamic sc
+//output: bool result
+export function isWebGPURenderValid(sc: DG.ScatterPlotViewer) {
+  if (scOriginalMarkerType == '')
+    scOriginalMarkerType = sc.props.markerType;
+  if (scOriginalMarkersColumnName == '')
+    scOriginalMarkersColumnName = sc.props.markersColumnName;
+  return sc.props.zoomAndFilter != 'pack and zoom by filter' 
+    && sc.props.markerType == scOriginalMarkerType
+    && sc.props.markersColumnName == scOriginalMarkersColumnName
+    && !sc.dataFrame.selection.anyTrue;
```

Comment:
I don't like this, I know I told you this idea, but now that I think of it, if scatterplot is preconfigured with marker col, and then the flag is turned on, this will not work. for now, as we only work with circles. the name of shape is circle, so just harcode the value of sc.props.markerType to be equal to DG.MARKER_TYPE.CIRCLE.
same with markerColumnName. make sure it is empty or empty string. `!sc.props.markersColumnName` will deal with both cases. and lose the global variables scOriginalMarkersColumnName and scOriginalMarkerType